### ISSUE TITLE
vm_compute chooses bizarre names for recursive functions in fix, when compute does not

### ISSUE THREAD

**Issue by ybertot** [coq/coq#10771]
#### Description of the problem
Compare the results of the following two commands:

    Compute (fun x => match x + 2 with S (S x) => x | _ => 0 end).

    Eval compute in (fun x => match x + 2 with S (S x) => x | _ => 0 end).

The symbol that appears after fix is `Ffix` in one case and `add` in the other case.  When explaining computation in Coq in an educational context, it is nicer to have the second result.  It is a pain to explain why there is a difference.

#### Coq Version
8.9.0

**Comment by ppedrot** [comment]
Unfortunately, I think it's not possible to retrieve this information out of the NbE algorithm, or at least it's going to be very costly. What do you like to get exactly?

**Comment by maximedenes** [comment]
I thought it would be easy to store the name of fixpoints, since we already store their types, don't we?

**Comment by ppedrot** [comment]
Definitely, but that's on the costly side, I think. People like @JasonGross won't be happy when the normalization of PHOAS trees is going to eat up more memory, they're already on a tight rope. Storing the names is at least one more word by fixpoint, YMMV.

**Comment by SkySkimmer** [comment]
We will need to store the relevances to fix #9779, so we may as well store a pointer to the Context.binder_annot.
